### PR TITLE
Use modern-ish jetty

### LINKED ISSUE #233: Upgrade jetty version
See #172.

Ensure that we can upgrade this (internal dependencies).

### PR DESCRIPTION
**Pull request by Ancient123**
Fixes #233 
Works towards #130

### CODE REVIEW

File: google-oauth-client-jetty/pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>com.google.oauth-client</groupId>
     <artifactId>google-oauth-client-parent</artifactId>
-    <version>1.30.4-SNAPSHOT</version><!-- {x-version-update:google-oauth-client:current} -->
+    <version>1.30.3</version><!-- {x-version-update:google-oauth-client:current} -->
```
Comment (elharo) [inline google-oauth-client-jetty/pom.xml]: I don't think this should have changed

File: google-oauth-client-jetty/pom.xml [hunk 2]
```diff
@@ -87,8 +87,9 @@
       <artifactId>google-oauth-client-java6</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.mortbay.jetty</groupId>
-      <artifactId>jetty</artifactId>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-server</artifactId>
+      <version>8.2.0.v20160908</version>
```
Comment (elharo) [inline google-oauth-client-jetty/pom.xml]: maybe a dependencyManagement section somewhere needs to be updated?

File: google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java [hunk 3]
```diff
@@ -118,7 +118,9 @@ public String getRedirectUri() throws IOException {
     server = new Server(port != -1 ? port : 0);
     Connector connector = server.getConnectors()[0];
     connector.setHost(host);
-    server.addHandler(new CallbackHandler());
+    //Deprecated
```
Comment (elharo) [inline google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java]: delete the commented lines

File: google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java [hunk 4]
```diff
@@ -254,7 +256,7 @@ public Builder setLandingPages(String successLandingPageUrl, String failureLandi
 
     @Override
     public void handle(
-        String target, HttpServletRequest request, HttpServletResponse response, int dispatch)
+        String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
```
Comment (elharo) [inline google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java]: This looks like a change in public API, which we are loath to do. Can the old method be retained and add this one, maybe one with a delegate to the other?

Comment (Ancient123) [reply google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java]: I have no idea how to to that! 
But, I figured after 4 years of there being a request to update a decade old highly vulnerable package ( [CVE-2011-4461][CVE-2017-7656][CVE-2017-7657][CVE-2017-7658][CVE-2017-9735][CVE-2019-10241][CVE-2019-10247] ) I might atleast post a PR to get some traction on the issue.

Comment (Ancient123) [reply google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java]: Also: 
This is public within `class CallbackHandler` which is not public. I am guessing that this is only exposed to the LocalServerReceiver class that it is nested within. Although my knowledge of java is garbage so I may be completely wrong.

File: google-oauth-client-jetty/pom.xml (hunk 1 above)
Comment (Ancient123) [reply google-oauth-client-jetty/pom.xml]: Fixed

File: google-oauth-client-jetty/pom.xml (hunk 2 above)
Comment (Ancient123) [reply google-oauth-client-jetty/pom.xml]: Found the upstream reference in the main pom.xml and fixed it there.

File: google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java (hunk 3 above)
Comment (Ancient123) [reply google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java]: Fixed.

File: google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java (hunk 4 above)
Comment (elharo) [reply google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java]: If the class is non-public, that's OK then.

Comment (Ancient123) [reply google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java]: Looks like there is one style issue I am creating. I will try and fix it. Please re-run kokoro.

**Comment by Ancient123** [comment]
Yay everything passes!

**Comment by chingor13** [comment]
We don't have this version of jetty internally (we do have 9.4.x) so we won't be able to import this in. We should likely update to 9.4 before we attempt to import it internally.

**Comment by Ancient123** [comment]
I am still looking at migrating to 9.4.x but the 8 => 9 upgrade is pretty non-trivial.

**Pull request merged**